File: mobile/src/screens/ReportsScreen.js
import React, { useState, useEffect } from 'react';
import { View, Text, FlatList, StyleSheet } from 'react-native';
import { Calendar } from 'react-native-calendars';
import { getReports } from '../api/api';

export default function ReportsScreen() {
  const [reportes, setReportes] = useState([]);
  const [fechaSeleccionada, setFechaSeleccionada] = useState(new Date().toISOString().split('T')[0]);

  const fetchReports = async (fecha) => {
    try {
      const response = await getReports(fecha);
      setReportes(response.data);
    } catch (error) {
      console.error(error);
    }
  };

  useEffect(() => {
    fetchReports(fechaSeleccionada);
  }, [fechaSeleccionada]);

  const renderItem = ({ item }) => {
    const nombreCompleto = item.personal 
      ? `${item.personal.nombres} ${item.personal.apellido_paterno} ${item.personal.apellido_materno}`
      : 'Desconocido';
      
    const horaEntrada = new Date(item.hora_entrada).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });
    const horaSalida = item.hora_salida ? new Date(item.hora_salida).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' }) : '--:--';

    return (
      <View style={styles.item}>
        <Text style={styles.name}>{nombreCompleto} - CI: {item.personal?.ci}</Text>
        <Text style={styles.times}>Entrada: {horaEntrada} | Salida: {horaSalida}</Text>
        <Text style={styles.status}>Estado: {item.estado}</Text>
      </View>
    );
  };

  return (
    <View style={styles.container}>
      <Calendar
        onDayPress={day => {
          setFechaSeleccionada(day.dateString);
        }}
        markedDates={{
          [fechaSeleccionada]: {selected: true, disableTouchEvent: true, selectedColor: '#00adf5'}
        }}
      />
      <Text style={styles.subtitle}>Reporte para: {fechaSeleccionada}</Text>
      <FlatList
        data={reportes}
        keyExtractor={item => item.id.toString()}
        renderItem={renderItem}
        ListEmptyComponent={<Text style={styles.empty}>No hay registros para esta fecha.</Text>}
        refreshing={false}
        onRefresh={() => fetchReports(fechaSeleccionada)}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: '#fff' },
  subtitle: { fontSize: 18, fontWeight: 'bold', marginVertical: 10, textAlign: 'center' },
  item: { backgroundColor: '#f0f0f0', padding: 15, marginHorizontal: 15, marginVertical: 8, borderRadius: 8, elevation: 2 },
  name: { fontSize: 16, fontWeight: 'bold' },
  times: { marginTop: 5, color: '#555' },
  status: { color: '#007BFF', fontWeight: 'bold', marginTop: 5 },
  empty: { textAlign: 'center', marginTop: 20, fontStyle: 'italic', color: '#888' }
});

File: flutter_app/lib/screens/reports_screen.dart
import 'package:flutter/material.dart';
import 'package:table_calendar/table_calendar.dart';
import '../services/api_service.dart';

class ReportsScreen extends StatefulWidget {
  const ReportsScreen({Key? key}) : super(key: key);

  @override
  State<ReportsScreen> createState() => _ReportsScreenState();
}

class _ReportsScreenState extends State<ReportsScreen> {
  DateTime _selectedDay = DateTime.now();
  DateTime _focusedDay = DateTime.now();
  List<dynamic> _reports = [];
  bool _isLoading = false;

  @override
  void initState() {
    super.initState();
    _fetchReports(_selectedDay);
  }

  void _fetchReports(DateTime date) async {
    setState(() => _isLoading = true);
    final dateString = "${date.year}-${date.month.toString().padLeft(2, '0')}-${date.day.toString().padLeft(2, '0')}";
    try {
      final reports = await ApiService.getReports(dateString);
      setState(() {
        _reports = reports;
      });
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(const SnackBar(content: Text('Error al cargar reportes')));
      }
    } finally {
      if (mounted) setState(() => _isLoading = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Reportes de Asistencia')),
      body: Column(
        children: [
          TableCalendar(
            locale: 'es_ES', // Calendario en español
            firstDay: DateTime.utc(2020, 1, 1),
            lastDay: DateTime.utc(2030, 12, 31),
            focusedDay: _focusedDay,
            selectedDayPredicate: (day) => isSameDay(_selectedDay, day),
            onDaySelected: (selectedDay, focusedDay) {
              setState(() {
                _selectedDay = selectedDay;
                _focusedDay = focusedDay;
              });
              _fetchReports(selectedDay);
            },
            calendarStyle: const CalendarStyle(
              selectedDecoration: BoxDecoration(color: Colors.blueAccent, shape: BoxShape.circle),
              todayDecoration: BoxDecoration(color: Colors.orangeAccent, shape: BoxShape.circle),
            ),
          ),
          const Divider(),
          Expanded(
            child: _isLoading 
                ? const Center(child: CircularProgressIndicator())
                : _reports.isEmpty
                    ? const Center(child: Text('No hay registros para esta fecha.', style: TextStyle(fontStyle: FontStyle.italic)))
                    : ListView.builder(
                        itemCount: _reports.length,
                        itemBuilder: (context, index) {
                          final item = _reports[index];
                          final personal = item['personal'] ?? {};
                          final nombreCompleto = "${personal['nombres'] ?? ''} ${personal['apellido_paterno'] ?? ''}".trim();
                          final ci = personal['ci'] ?? 'N/A';
                          
                          final dtEntrada = DateTime.tryParse(item['hora_entrada'] ?? '');
                          final dtSalida = item['hora_salida'] != null ? DateTime.tryParse(item['hora_salida']) : null;
                          
                          final strEntrada = dtEntrada != null ? "${dtEntrada.toLocal().hour.toString().padLeft(2,'0')}:${dtEntrada.toLocal().minute.toString().padLeft(2,'0')}" : '--:--';
                          final strSalida = dtSalida != null ? "${dtSalida.toLocal().hour.toString().padLeft(2,'0')}:${dtSalida.toLocal().minute.toString().padLeft(2,'0')}" : '--:--';

                          return Card(
                            margin: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
                            child: ListTile(
                              leading: const CircleAvatar(backgroundColor: Colors.blueAccent, child: Icon(Icons.person, color: Colors.white)),
                              title: Text('$nombreCompleto (CI: $ci)', style: const TextStyle(fontWeight: FontWeight.bold)),
                              subtitle: Text('Entrada: $strEntrada | Salida: $strSalida\nEstado: ${item['estado']}'),
                              isThreeLine: true,
                            ),
                          );
                        },
                      ),
          ),
        ],
      ),
    );
  }
}
